### PR TITLE
Audio

### PR DESCRIPTION
**Pull request by laurynbassett**

### CODE REVIEW

File: components/Audio.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+import { FontAwesome } from "@expo/vector-icons";
+import { Audio } from "expo-av";
+import * as Permissions from "expo-permissions";
+import * as FileSystem from "expo-file-system";
+
+import Layout from "../constants/Layout";
+import { fetchMessages, postMessage, postAudio } from "../store";
+import { formatText, recordingCallback, setAudioMode } from "../utils";
+
+class SingleChat extends Component {
```
Comment (israkhan) [inline components/Audio.js:14]: should the component name be SingleChat or Audio?

File: components/Audio.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+	sendMessageError: state.messages.sendMessageError
+});
+
+const mapDispatch = dispatch => ({
+	fetchMessages: () => dispatch(fetchMessages()),
+	postMessage: msg => dispatch(postMessage(msg)),
+	postAudio: (file, text) => dispatch(postAudio(file, text))
+});
+
+export default connect(mapState, mapDispatch)(SingleChat);
```
Comment (israkhan) [inline components/Audio.js:246]: Change name?

File: components/SingleChat.js [hunk 3]
```diff
@@ -39,124 +65,386 @@ class SingleChat extends Component {
...
         <Text style={{ color: "red" }}>{this.props.sendMessageError}</Text>
         <GiftedChat
           messages={this.props.messages}
           user={{
             _id: this.props.uid,
-            name: this.props.displayName,
+            name: this.props.displayName
           }}
-          onSend={this.handleSendMessage}
+          onSend={this.handleSendMessage.bind(this)}
```
Comment (israkhan) [inline components/SingleChat.js:411]: does "bind" need to be called here if it's already being called in the constructor? same with line 412

File: components/SingleChat.js [hunk 4]
```diff
@@ -1,26 +1,49 @@
...
+      audioPermission: false,
+      audioUrl: null,
+      isRecording: false,
+      isRecordingPlaying: false,
+      isAudioPlaying: false,
+      isLoading: false,
+      playbackInstancePosition: null,
+      playbackInstanceDuration: null
     };
+    this.recording = null;
```
Comment (israkhan) [inline components/SingleChat.js:28]: A lot of these Audio methods are in the Audio component already... is there any way to reduce the code in this file? There seem to be a lot of repetition between this file and Audio.js

Comment (laurynbassett) [reply components/SingleChat.js:28]: I meant to delete Audio. I just copy/pasted what I previously had in Single Chat there so it was easier for me to revise but will remove.

**Pull request closed without merging**